model/task/task_test.go

```diff
@@ -2027,46 +2027,76 @@ func TestTopologicalSort(t *testing.T) {
 }
 
 func TestActivateTasks(t *testing.T) {
-	require.NoError(t, db.ClearCollections(Collection, event.EventCollection))
+	defer func() {
+		assert.NoError(t, db.ClearCollections(Collection, event.EventCollection))
+	}()
 
-	tasks := []Task{
-		{Id: "t0", Priority: evergreen.DisabledTaskPriority},
-		{Id: "t1", DependsOn: []Dependency{{TaskId: "t0"}}, Activated: false},
-		{Id: "t2", DependsOn: []Dependency{{TaskId: "t0"}, {TaskId: "t1"}}, Activated: false, DeactivatedForDependency: true},
-		{Id: "t3", DependsOn: []Dependency{{TaskId: "t0"}}, Activated: false, DeactivatedForDependency: true},
-		{Id: "t4", DependsOn: []Dependency{{TaskId: "t0"}, {TaskId: "t3"}}, Activated: false, DeactivatedForDependency: true},
-		{Id: "t5", Activated: true},
-	}
-	for _, task := range tasks {
-		require.NoError(t, task.Insert())
-	}
+	t.Run("DependencyChain", func(t *testing.T) {
+		require.NoError(t, db.ClearCollections(Collection, event.EventCollection))
+		tasks := []Task{
+			{Id: "t0", Priority: evergreen.DisabledTaskPriority},
+			{Id: "t1", DependsOn: []Dependency{{TaskId: "t0"}}, Activated: false},
+			{Id: "t2", DependsOn: []Dependency{{TaskId: "t0"}, {TaskId: "t1"}}, Activated: false, DeactivatedForDependency: true},
+			{Id: "t3", DependsOn: []Dependency{{TaskId: "t0"}}, Activated: false, DeactivatedForDependency: true},
+			{Id: "t4", DependsOn: []Dependency{{TaskId: "t0"}, {TaskId: "t3"}}, Activated: false, DeactivatedForDependency: true},
+			{Id: "t5", DependsOn: []Dependency{{TaskId: "t0"}}, Activated: true, DeactivatedForDependency: true},
+		}
+		for _, task := range tasks {
+			require.NoError(t, task.Insert())
+		}
 
-	updatedIDs := []string{"t0", "t3", "t4"}
-	err := ActivateTasks([]Task{tasks[0]}, time.Time{}, true, "")
-	assert.NoError(t, err)
+		updatedIDs := []string{"t0", "t3", "t4"}
+		err := ActivateTasks([]Task{tasks[0]}, time.Time{}, true, "")
+		assert.NoError(t, err)
 
-	dbTasks, err := FindAll(All)
-	assert.NoError(t, err)
-	assert.Len(t, dbTasks, 6)
+		dbTasks, err := FindAll(All)
+		assert.NoError(t, err)
+		assert.Len(t, dbTasks, 6)
 
-	for _, task := range dbTasks {
-		assert.Equal(t, task.Priority, int64(0))
-		if utility.StringSliceContains(updatedIDs, task.Id) {
-			assert.True(t, task.Activated)
-			events, err := event.FindAllByResourceID(task.Id)
-			require.NoError(t, err)
-			require.Len(t, events, 1)
-		} else {
-			for _, origTask := range tasks {
-				if origTask.Id == task.Id {
-					assert.Equal(t, origTask.Activated, task.Activated, fmt.Sprintf("task '%s' mismatch", task.Id))
+		for _, task := range dbTasks {
+			assert.Equal(t, task.Priority, int64(0))
```

Comment:
nit: the order for `Equal` is expected value, actual value, so the two args should be reversed. I imagine you could also just `assert.EqualValues` so you don't have to explicitly make an `int64`, but it's not that important.

Reply:
done and done